### DISCUSSION TITLE
Now playing for your website updated

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#7138] [Show and tell]
AzuraCast recently changed to a newer Centrifugo version, which can affect "Now playing" scripts that use Server-Sent Events (SSE).

I have updated my well-known [AzuraCast HPNP (High-Performance Now Playing) example for station websites, using SSE (Server-Sent Events)](https://gist.github.com/Moonbase59/d42f411e10aff6dc58694699010307aa) so it should work with _both_ older and newer AzuraCast releases.

**EDIT:** Now includes @PeWe79’s improvement, see below.

- Source code: https://gist.github.com/Moonbase59/d42f411e10aff6dc58694699010307aa
- Documentation: https://gist.github.com/Moonbase59/d42f411e10aff6dc58694699010307aa?permalink_comment_id=4786603#gistcomment-4786603
- Live demo: https://grav1.niteradio.net/wh/sse_cf_demo.html

**Comment by LordHelmchen666** [comment]
@Moonbase59 When i try with demo.azuracast.com it works fine. As soon as i use my own station (also tested with niteradio) nothing is shown, but it seems to connect. Could i miss any setting?
![Screenshot_20240523_161921](https://github.com/AzuraCast/AzuraCast/assets/43477041/be333eb7-8662-4470-bf3b-a1133e22d655)
![Screenshot_20240523_162047](https://github.com/AzuraCast/AzuraCast/assets/43477041/449fb25f-4788-42ee-98f7-af595cba9518)

**Reply by Moonbase59** [reply]
Did you—per chance—use the "azuratest-radio" HTML classes, i.e. by just changing the subscriptions in the `sse_cf_demo.js` but _not_ in the HTML?

(The HTML must use station-specific class names, since you could display info for _more than one station_.)

I just tested again with AzuraCast RR #3d27d0f (2024-05-22 14:01) and it works fine here.

I was wondering because the heading in your second screenshot shows "Station Name", so maybe it’s still using
```
<h2 class="np-azuratest-radio-station-name">Station Name</h2>
```
instead of (I think)
```
<h2 class="np-eisradio-station-name">Station Name</h2>
```
You could use a simple search-and-replace in the HTML: Change all `np-azuratest-radio-` to `np-eisradio-` (assuming your station is called `eisradio`).

**Reply by LordHelmchen666** [reply]
Shame on me. You are totally right. Sometimes you don't see the obvious mistakes. Thank you so much!
Perhaps you could point out somewhere that we need to change that too.

**Reply by Moonbase59** [reply]
Such things happen ;-) It’s in section 4 of the [installation instructions](https://gist.github.com/Moonbase59/d42f411e10aff6dc58694699010307aa?permalink_comment_id=4786603#gistcomment-4786603).

**Reply by LordHelmchen666** [reply]
Ready to show it working. https://www.eisradio.de/

**Reply by PeWe79** [reply]
Fix error console log on Firefox :
```
The connection to https://demo.azuracast.com/api/live/nowplaying/sse?cf_connect={%22subs%22:{%22station:azuratest_radio%22:{%22recover%22:true,%22timezone%22:%22Etc/UTC%22}}} was interrupted while the page was loading.
```
Add this code under ```function initEvents()``` sse_cf_demo.js :
```
window.addEventListener("beforeunload", function () {
      if (evtSource !== null) {
        evtSource.close();
        evtSource = null;
      }
 });
```
Thanks.

**Reply by Moonbase59** [reply]
@PeWe79 Thanks for the suggestion!

It _does_ make things cleaner when refreshing the page, but the `beforeunload` event suffers from some problems:

- It is not reliably fired, especially on mobile platforms.
- In Firefox, it’s not compatible with the back/forward cache, meaning that Firefox won’t include a page in the bfcache when it has `beforeunload` listeners, which can affect performance.

I included it nevertheless, should any dramatic performance issues arise, let me know.

**Comment by ebinvthomas** [comment]
Thank you for the fix. 

Ucartz is now planning to offer managed support for Azuracast. I hope we can contribute more to Azuracast's continues development.
